src/d/actor/d_a_kui.cpp

```diff
@@ -4,68 +4,541 @@
+            }
+
+            if (i_this->field_0x2DC == 0 && actor->health == 3 && REG0_S(3) == 0) {
+                dComIfGs_onSwitch(i_this->mSwitchNo, fopAcM_GetRoomNo(actor));
+            }
+        }
+
+        cLib_addCalcAngleS2(&i_this->field_0x2E2, i_this->field_0x2E4, 4, i_this->field_0x2E6);
+        cLib_addCalcAngleS2(&i_this->field_0x2E4, 0, 1, REG17_S(6) + 0x80);
+        cLib_addCalcAngleS2(&i_this->field_0x2E6, REG17_S(7) + 0x100, 1, REG17_S(8) + 0x40);
+
+        MtxTrans(actor->home.pos.x, actor->home.pos.y, actor->home.pos.z, FALSE);
+        mDoMtx_YrotM(*calc_mtx, actor->current.angle.y);
+
+        MtxPush();
+        mDoMtx_YrotM(*calc_mtx, actor->shape_angle.y);
+        mDoMtx_XrotM(*calc_mtx, x + i_this->field_0x2E2);
+        mDoMtx_ZrotM(*calc_mtx, z);
+        mDoMtx_YrotM(*calc_mtx, -actor->shape_angle.y);
+        MtxScale(unk_f, unk_f, unk_f, TRUE);
+        i_this->mpModel->setBaseTRMtx(*calc_mtx);
+
+
+        MtxPull();
+        mDoMtx_YrotM(*calc_mtx, actor->shape_angle.y + REG0_S(5));
+        mDoMtx_XrotM(*calc_mtx, actor->current.angle.x + REG0_S(6));
+        mDoMtx_YrotM(*calc_mtx, -(actor->shape_angle.y + REG0_S(5) + 0x4000));
+
+        MtxScale(unk_f, 1.0, unk_f, TRUE);
+        i_this->mpModel2->setBaseTRMtx(*calc_mtx);
+        MtxTrans(0.0, REG0_F(6) + -850.0f, 0.0, TRUE);
+
+        temp2.setall(0.0f);
+        MtxPosition(&temp2, &actor->current.pos);
+
+        MtxTrans(actor->current.pos.x, actor->current.pos.y, actor->current.pos.z, FALSE);
+        mDoMtx_YrotM(*calc_mtx, actor->current.angle.y);
+        MtxScale(actor->scale.x, actor->scale.y, actor->scale.z, TRUE);
+        cMtx_copy(*calc_mtx, i_this->field_0x2A8);
+        i_this->field_0x2D8->Move();
+    } else {
+        MtxTrans(actor->current.pos.x, actor->current.pos.y, actor->current.pos.z, FALSE);
+        mDoMtx_YrotM(*calc_mtx, actor->current.angle.y);
+        if (i_this->type == 2 || i_this->type == 4) {
+            i_this->mpModel->setBaseTRMtx(*calc_mtx);
+        }
+
+        i_this->mpModel2->setBaseScale(actor->scale);
+        i_this->mpModel2->setBaseTRMtx(*calc_mtx);
+        if (i_this->field_0x2A2) {
+            cMtx_scale(local_mtx, 4.0f, 4.0f, 4.0f);
+        } else {
+            cMtx_scale(local_mtx, actor->scale.x, actor->scale.y, actor->scale.z);
+        }
+
+        cMtx_concat(*calc_mtx, local_mtx, i_this->field_0x2A8);
+        i_this->field_0x2D8->Move();
+    }
+
+    if (i_this->type == 2 || i_this->type == 4) {
+        BOOL is_switch = dComIfGs_isSwitch(i_this->mSwitchNo, dComIfGp_roomControl_getStayNo());
+
+        if (!is_switch) {
+            if (actor->health == 3 && i_this->field_0x308 == 0) {
+                i_this->field_0x308 = 1000;
+            }            
+        } else {
+            actor->current.pos.y = actor->home.pos.y - 70.0f;
+        }
+
+        if (i_this->field_0x308 != 0) {
+            i_this->field_0x308--;
+
+            s16 finished = REG8_S(3) + 0x3CA;
```

Comment:
```suggestion
            s16 finished = REG8_S(3) + 970;
```
This can be written in decimal since it's probably a timer value for something (like the above 1000 is also in decimal)